recipes/stx/all/conanfile.py

```diff
@@ -0,0 +1,153 @@
+                compiler.libcxx == 'libc++' and
+                compiler_version < 10):
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which clang < 10 with libc++ lacks'
+            )
+
+        if compiler == 'apple-clang':
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which apple-clang with libc++ lacks'
+            )
+
+        if (compiler == 'Visual Studio' and self.options.shared and
+                tools.Version(self.version) <= '1.0.1'):
+            raise ConanInvalidConfiguration(
+                'shared library build does not work on windows with '
+                'STX version <= 1.0.1'
+            )
+
+    def requirements(self):
+        if self.options.backtrace:
+            self.requires('abseil/20200923.1')
+
+    def source(self):
+        tools.get(**self.conan_data['sources'][self.version])
+        tools.rename('STX-{}'.format(self.version), dst=self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data.get('patches', {}).get(self.version, []):
+            tools.patch(**patch)
+
+        cmake = CMake(self)
+        cmake.definitions['STX_BUILD_SHARED'] = self.options.shared
+        cmake.definitions['STX_ENABLE_BACKTRACE'] = self.options.backtrace
+        cmake.definitions['STX_ENABLE_PANIC_BACKTRACE'] = \
+            self.options.panic_handler == 'backtrace'
+        cmake.definitions['STX_OVERRIDE_PANIC_HANDLER'] = \
+            self.options.panic_handler == None
+        cmake.definitions['STX_VISIBLE_PANIC_HOOK'] = \
+            self.options.visible_panic_hook
+
+        cmake.configure(build_folder=self._build_subfolder)
+        cmake.build()
+
+    def package(self):
+        self.copy(
+            '*.h',
+            dst='include',
+            src=os.path.join(self._source_subfolder, 'include')
+        )
+
+        self.copy('*.lib', dst='lib', keep_path=False)
+        self.copy('*.dll', dst='bin', keep_path=False)
+        self.copy('*.so', dst='lib', keep_path=False)
+        self.copy('*.dylib', dst='lib', keep_path=False)
+        self.copy('*.a', dst='lib', keep_path=False)
+
+        self.copy('LICENSE', dst='licenses', src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+
+        if self.options.backtrace:
+            self.cpp_info.requires = [
+                'abseil::absl_stacktrace',
+                'abseil::absl_symbolize'
+            ]
+
+        if self.options.visible_panic_hook:
+            self.cpp_info.defines.append('STX_VISIBLE_PANIC_HOOK')
+
+        if self.options.panic_handler == None:
+            self.cpp_info.defines.append('STX_OVERRIDE_PANIC_HANDLER')
+
+        if self.options.panic_handler == 'backtrace':
+            self.cpp_info.defines.append('STX_ENABLE_PANIC_BACKTRACE')
```

Comment:
I guess that atomic should be added in system libs if Android? https://github.com/lamarrr/STX/blob/539fa551987e8a5e15700c6df06d87437e9af356/CMakeLists.txt#L218

Reply:
Good point, I missed that.